### PR TITLE
Automatic, PEP386-compliant version number.

### PR DESCRIPTION
**Pull request by pberkes**
Based on John Wiggin's work for traits, codetools, etc.

### CODE REVIEW

File: chaco/__init__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 """ Two-dimensional plotting application toolkit.
     Part of the Chaco project of the Enthought Tool Suite.
 """
-__version__ = '4.6.0-dev'
+from ._version import full_version as __version__
```
Comment (itziakos) [inline chaco/__init__.py]: flake8 complains about `__version__` imported but not used. Should we include an `__all__` definition?

File: setup.py [hunk 2]
```diff
@@ -1,77 +1,174 @@
...
+    with open(filename, "wt") as fp:
+        fp.write(template.format(version=VERSION,
+                                 full_version=fullversion,
+                                 git_revision=git_rev,
+                                 is_released=IS_RELEASED))
+
+
+if __name__ == "__main__":
+    write_version_py()
+    from chaco import __requires__, __version__
```
Comment (itziakos) [inline setup.py]: This looks like an old style relative import. I think it will cause problems with python 3 in the future.

Comment (rkern) [inline setup.py]: How so? setup.py isn't in a package.

Comment (itziakos) [inline setup.py]: Yup, you are right.  My mistake. Please ignore my comment.

File: chaco/__init__.py (hunk 1 above)
Comment (pberkes) [inline chaco/__init__.py]: I vote for a mere `# noqa`

**Comment by itziakos** [comment]
:+1:  to me, just minor comment.

**Comment by itziakos** [comment]
The travis-ci failure however, looks fun

File: chaco/__init__.py (hunk 1 above)
Comment (itziakos) [inline chaco/__init__.py]: As long as flake8 does not complain I am ok.

File: dev_requirements.txt [hunk 3]
```diff
@@ -6,3 +6,4 @@ coverage
 -e git+http://github.com/enthought/traits.git#egg=traits
 -e git+http://github.com/enthought/traitsui.git#egg=traitsui
 -e git+http://github.com/enthought/enable.git#egg=enable
+-e git+http://github.com/nucleic/kiwi.git#egg=kiwisolver
```
Comment (itziakos) [inline dev_requirements.txt]: I would suggest to use the offcial pypi egg for kiwisolver

**Comment by itziakos** [comment]
updated converalls.io config to only use the github status api

**Comment by pberkes** [comment]
Please don't merge yet, pending better git tag matching string.

**Comment by pberkes** [comment]
As far as I'm concerned, this is ready for a final review.

File: setup.py [hunk 4]
```diff
@@ -1,77 +1,181 @@
...
+        env['LC_ALL'] = 'C'
+        out = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, env=env,
+        ).communicate()[0]
+        return out
+
+    try:
+        out = _minimal_ext_cmd(
+            ['git', 'describe', '--match', '[0-9]*.[0-9]*.[0-9]*', '--tags'])
+    except OSError:
```
Comment (jonathanrocher) [inline setup.py]: Could you add a comment with what you are expecting here?

Comment (pberkes) [inline setup.py]: Good point, I will

File: setup.py (hunk 2 above)
Comment (pberkes) [inline setup.py]: Comment ignored.
;-)

File: setup.py (hunk 4 above)
Comment (pberkes) [inline setup.py]: Done

**Comment by pberkes** [comment]
@jonathanrocher Any other comment?

**Comment by jonathanrocher** [comment]
Sorry. LGTM. Merging.

**Comment by mdickinson** [comment]
For some reason, since this PR was merged, a `python setup.py develop` hangs when trying to install Chaco's dependencies, entering some kind of infinite memory-hungry loop.  I've reproduced this both with and without Canopy, on Linux and on OS X.  Can anyone else reproduce?

**Pull request merged**